### DISCUSSION TITLE
Run nornir without config file

### DISCUSSION THREAD

**Discussion by Sivolen** [nornir-automation/nornir#799] [Q&A]
Hi!

Please help me, how to run nornir without config file
I created dict 
```
        host = {
                  "name": "core",
                  "connection_options": {},
                  "hostname": "10.255.100.1",
                  "port": 22,
                  # "username": null,
                  # "password": null,
                  "platform": "ce"
nr_driver = InitNornir(inventory=host, logging=self.logging_file)
```
but it doesn't work
if i use next dict he also doesn't' work
```
inventory_cfg = {'yzh-krm24-cr420-r01': {'connection_options': {},
                                   'data': {
                                   'hostname': '10.255.100.81',
                                   'name': 'yzh-krm24-cr420-r01',
                                   'password': None,
                                   'platform': 'ce',
                                   'port': None,
                                   'username': None}}}}
nr_driver = InitNornir(inventory=inventory_cfg, logging=self.logging_file)

```

**Comment by dbarrosop** [comment]
You can't create inventories from dicts like that, you need proper objects. I'd suggest you to create a plugin that does that, shouldn't be too hard to adapt the [`SimpleInventory`](https://github.com/nornir-automation/nornir/blob/main/nornir/plugins/inventory/simple.py) plugin to do that.

**Reply by Sivolen** [reply]
> You can't create inventories from dicts like that, you need proper objects. I'd suggest you to create a plugin that does that, shouldn't be too hard to adapt the [`SimpleInventory`](https://github.com/nornir-automation/nornir/blob/main/nornir/plugins/inventory/simple.py) plugin to do that.

Thanks!
I found the nornir-sql plugin, it suits me just right
